pkgs/development/python-modules/pyqt/6.x.nix

```diff
@@ -0,0 +1,134 @@
+{ lib
+, buildPythonPackage
+, isPy27
+, fetchPypi
+, pkg-config
+, dbus
+, lndir
+, dbus-python
+, sip
+, pyqt6-sip
+, pyqt-builder
+, qt6Packages
+, withConnectivity ? false
+, withMultimedia ? false
+, withWebKit ? false
+, withWebSockets ? false
+, withLocation ? false
+}:
+
+buildPythonPackage rec {
+  pname = "PyQt6";
+  version = "6.3.1";
+  format = "pyproject";
+
+  disabled = isPy27;
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-jMbiHbr3BH0fyJfjlszZcQoS8u+XZWPa1l8GAX0sl1c=";
+  };
+
+  patches = [
+    # Fix some wrong assumptions by ./project.py
+    # TODO: figure out how to send this upstream
+    # ./pyqt5-fix-dbus-mainloop-support.patch
+    # confirm license when installing via pyqt6_sip
+    ./pyqt5-confirm-license.patch
+  ];
+
+  # be more verbose
+  postPatch = ''
+    cat >> pyproject.toml <<EOF
+    [tool.sip.project]
+    verbose = true
+    EOF
+  '';
+
+  postUnpack = ''
+  echo QMAKEPATH=$QMAKEPATH
+
+  # paralellize compilation of , which happens within ./configure
+  export MAKEFLAGS+=" -j$NIX_BUILD_CORES"
+  '';
+
+  outputs = [ "out" "dev" ];
+
+  enableParallelBuilding = true;
+
+  dontWrapQtApps = true;
+
+  nativeBuildInputs = with qt6Packages; [
+    pkg-config
+    lndir
+    sip
+    qtbase
+    qtsvg
+    qtdeclarative
+    qtwebchannel
+    qmake
+    qtquick3d
+    qtquicktimeline
+  ]
+    ++ lib.optional withConnectivity qtconnectivity
+    ++ lib.optional withMultimedia qtmultimedia
+    ++ lib.optional withWebKit qtwebkit
+    ++ lib.optional withWebSockets qtwebsockets
+    ++ lib.optional withLocation qtlocation
+  ;
```

Comment:
Isn't specifying them once in `buildInputs` enough?

Reply:
I think it's correct the way it is.

The Qt module deps for PyQt6 have things needed both on the native platform and on the destination platform, as at build time it needs to look through the specs in the dev output to generate code.

The 5.x version of this package does the same thing.

Example output when not using `nativeBuildInputs`:

```
python3.10-PyQt6>   /nix/store/f5hlylpngpc59nnh6knxn70g20p0vhza-qtbase-6.3.1-dev/bin/qmake QtQml.pro
python3.10-PyQt6>   Info: creating stash file /build/tmpemfmvvwj/cfgtest_QtQml/.qmake.stash
python3.10-PyQt6>   Project ERROR: Unknown module(s) in QT: qml
```